src/smt/smt_engine.cpp

```diff
@@ -2319,7 +2314,25 @@ void SmtEngine::setDefaults() {
                << std::endl;
       options::bitvectorInequalitySolver.set(false);
     }
+    if (options::bvLazyRewriteExtf())
+    {
+      if (options::bvLazyRewriteExtf.wasSetByUser())
+      {
+        throw OptionException(
+            "--bv-lazy-rewrite-extf is not supported with proofs");
+      }
+      Notice() << "SmtEngine: turning off bv lazy rewrite extf to support proofs"
+               << std::endl;
+      options::bvLazyRewriteExtf.set(false);
+    }
   }
+
+  if( !options::bitvectorEqualitySolver() ){
```

Comment:
But 2317--2327 should run only if proofs are enabled, while 2330--2333 should always run.

Reply:
Well, 2317--2327 should always run when !options::bitvectorEqualitySolver() because it is nicer to not just silently disable something that the user set.

Reply:
gotcha